app/actions/remote/entry/common.ts

```diff
@@ -191,14 +194,49 @@ const fetchAppEntryData = async (serverUrl: string, sinceArg: number, initialTea
     // Fetch in parallel teams / team membership / channels for current team / user preferences / user
     const promises: [Promise<MyTeamsRequest>, Promise<MyChannelsRequest | undefined>, Promise<MyUserRequest>] = [
         fetchMyTeams(serverUrl, fetchOnly),
-        initialTeamId ? fetchMyChannelsForTeam(serverUrl, initialTeamId, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled) : Promise.resolve(undefined),
+        onLoadTeamId ? fetchMyChannelsForTeam(serverUrl, onLoadTeamId, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled) : Promise.resolve(undefined),
         fetchMe(serverUrl, fetchOnly),
     ];
 
     const resolution = await Promise.all(promises);
     const [teamData, , meData] = resolution;
     let [, chData] = resolution;
 
+    let initialTeamId = onLoadTeamId;
+    let initialChannelId = channelId;
+    let gmConverted = false;
+
+    if (channelId && chData?.channels) {
+        // check if channelId is in list of team's channels returned by server
+        const channelInServerData = chData.channels.find((channel) => channel.id === channelId);
+
+        // if channel is not found in server data, we need to check if it
+        // was a GM that is converted to a private or public channel in a different team than
+        // the team the mobile app was last closed in.
+        // Although yon can convert GM only to a pirvate channel, a private channel can furthur be converted to a public channel.
+        // So between the mobile app being on the GM and reconnecting,
+        // it may have become either a public or a private channel. So we need to check for both.
+        if (!channelInServerData) {
+            const existingChannel = await getChannelById(database, channelId);
+            if (existingChannel && existingChannel.type === General.GM_CHANNEL) {
+                // Okay, so now we know the channel existsin in mobile app's database as a GM.
+                // We now need to also check if channel on server is actually a private channel,
+                // and if so, which team does it belong to now. That team will become the
+                // active team on mobile app after this point.
+
+                const client = NetworkManager.getClient(serverUrl);
+                const serverChannel = await client.getChannel(channelId);
+                if (serverChannel.type === General.PRIVATE_CHANNEL || serverChannel.type === General.OPEN_CHANNEL) {
+                    initialTeamId = serverChannel.team_id;
+                    initialChannelId = channelId;
+                    gmConverted = true;
+
+                    chData = await fetchMyChannelsForTeam(serverUrl, serverChannel.team_id, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled);
```

Comment:
if it is the same team as the current team would you need to execute this again? I think not as they should be already in `chData` right ?

Reply:
Correct, its already in chData.